TST: fixturize series/test_alter_axes.py

h-vetinari opened the pull request:
- [x] prep for #22225, in the sense that it preempts (and splits off) the test-related changes that have been required on the DataFrame-side of the PR (see  #22236)
- [x] tests modified / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

In particular, it takes the fixture-like attributes of `tests/series/common.TestData` and transforms them into fixtures in `tests/series/conftest.py`, with the eventual goal of replacing all the `TestData`-attributes in the Series tests, similar to #22471 (I can also open a sister issue to that for the Series tests).

h-vetinari committed:
TST: fixturize series/test_alter_axes.py

h-vetinari commented:
@jreback 
> can you add doc-strings and meaningful names for these. (e.g. ts -> datetime_series)

This is the same issue as in #22236 - these "fixture-like" `TestData`-attributes will need to be replaced in all Series tests, and renaming them now will make it harder to simply replace them (e.g. if someone were to work on a different module, to know that `self.series` needs to be replaced with `string_series`).

One possible approach to satisfy both demands would be to open up a sister issue to #22471 for Series, and have a "translation guide" like
* `self.ts` -> `datetime_series`
* `self.series` -> `string_series
* `self.objSeries` -> `object_series`
* `self.empty` -> `empty_series`

The same approach could be applied to #22236 / #22471 if you're unhappy with the fixture names there (cc @gfyoung).

h-vetinari committed:
Review (jreback)

h-vetinari commented:
@jreback

I added the "translations" as requested - it's gonna make transitioning the other modules a fair bit harder, but for only four fixtures, I think it's doable.

jreback commented on pandas/tests/series/conftest.py:
```diff
@@ -0,0 +1,31 @@
+from pandas import Series
+
+
+@pytest.fixture
```
see pandas/conftest.py for examples, yes that might be ok, the point is new features need a doc-string

jreback reviewed (changes requested):
doc comments, ping on green.

pep8speaks commented:
Hello @h-vetinari! Thanks for updating the PR.

 - In the file [`pandas/tests/series/conftest.py`](https://github.com/pandas-dev/pandas/blob/cb1270339c363bf3b4cc28ee1f6d2000dd0babb8/pandas/tests/series/conftest.py), following are the PEP8 issues :

> [Line 12:1](https://github.com/pandas-dev/pandas/blob/cb1270339c363bf3b4cc28ee1f6d2000dd0babb8/pandas/tests/series/conftest.py#L12): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 24:1](https://github.com/pandas-dev/pandas/blob/cb1270339c363bf3b4cc28ee1f6d2000dd0babb8/pandas/tests/series/conftest.py#L24): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 36:1](https://github.com/pandas-dev/pandas/blob/cb1270339c363bf3b4cc28ee1f6d2000dd0babb8/pandas/tests/series/conftest.py#L36): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace

h-vetinari commented:
@jreback 

All green. This is very similar to #22236 - can I please ask you to take a look there again? I incorporated all your feedback (fixturize, split off warnings into new PR, etc.), and don't know what's left to do - it's been a very drawn-out process to eke out further instructions for what could have been a simple test clean-up.

Based on this review here, I now also added docstrings to the fixtures there; I didn't rename them yet (but added a note in each fixture), because otherwise the fixturization of the other modules would be that much harder (and it's much easier to run a simple search and replace once everything is translated, i.e. #22471 is done).

jreback commented:
it’s long and drawn out because th we PRs have too much inrelated changes in them

1 thing per PR will get u merged much faster with less comments 

even things as simple as changing unrelated whatsnew notes

the more code the more i have to go over with a fine toothed comb 

further responding to my comments with further questions - while for sure ask if things are unclear
can certainly make things take longer

h-vetinari commented:
@jreback anything left to do on this. Someone wants to already start working on fixturising the rest of the series tests, see #22550

jreback commented on pandas/tests/series/conftest.py:
```diff
@@ -0,0 +1,49 @@
+    """
+    Fixture for Series of floats with DatetimeIndex
+
+    See pandas.util.testing.makeTimeSeries
```
I don't think these See .. comments make sense here, just keep the summary line.

h-vetinari replied:
Removed those comments

jreback reviewed (changes requested):
small comments ping on green. 

@h-vetinari the pings are fine, but if someone wants to work on things based on this nothing is stopping them, they can rebase off of this PR, this is a quite common workflow. This will be merged after all comments are satisfied. Pls be patient. We have a tremendous amount of PRs

h-vetinari committed:
Review (jreback)

h-vetinari commented:
@jreback 

> @h-vetinari the pings are fine, but if someone wants to work on things based on this nothing is stopping them, they can rebase off of this PR, this is a quite common workflow. This will be merged after all comments are satisfied. Pls be patient. We have a tremendous amount of PRs

Yeah, I get there's lots of traffic. I mentioned this because this PR was basically finished and it's not a "beginner workflow" to rebase off an open PR.

h-vetinari commented:
@jreback 
> small comments ping on green.

pint on green

jreback commented:
thanks!
